frontend/src/components/util/routeInjector.tsx

```diff
@@ -7,47 +7,74 @@ import { getAllEditionsFromPage, getEditionByName, getEditionOnUrl } from "../..
 import { useEffect } from "react";
 import applicationPaths from "../../properties/applicationPaths";
 import useTranslation from "next-translate/useTranslation";
+import { useCurrentUser } from "../../hooks/useCurrentUser";
 
 export default function RouteInjector({ children }: any) {
+    const { error: userError } = useCurrentUser(true);
+    const userIsLoggedIn: boolean = !userError;
+
     const { t } = useTranslation("common");
     const [contextEditionUrl, setContextEditionUrl] = useEdition();
-    const { data: contextEdition, error: contextEditionError } = useSWR(contextEditionUrl, getEditionOnUrl);
+    console.log(contextEditionUrl);
+    const { data: contextEdition, error: contextEditionError } = useSWR(
+        userIsLoggedIn ? contextEditionUrl : null,
+        getEditionOnUrl
+    );
 
     const contextEditionName = contextEdition?.name;
+    const hadContextError = !!contextEditionError;
 
     const router = useRouter();
     const replace = router.replace;
     const query = router.query as { edition?: string };
     const routerEditionName = query.edition;
     const { data: fetchedRouterEdition, error: fetchedEditionError } = useSWR(
-        routerEditionName,
+        userIsLoggedIn ? routerEditionName : null,
         getEditionByName
     );
     const fetchedRouterEditionUrl = fetchedRouterEdition?._links?.self?.href;
+    const hadRouterError = !!fetchedEditionError;
 
     const { data: availableEditions, error: availableEditionsError } = useSWR(
-        !contextEditionUrl && !routerEditionName
+        !contextEditionUrl && !routerEditionName && userIsLoggedIn
             ? getQueryUrlFromParams(apiPaths.editions, { sort: "year" })
             : null,
         getAllEditionsFromPage
     );
     const latestEdition = availableEditions?.at(0);
     const latestEditionName = latestEdition?.name;
 
+    console.log(contextEditionError || fetchedEditionError || availableEditionsError);
```

Comment:
This always gets logged to the console, so this should probably go

Reply:
Def!